src/util/NamedColor.cpp

```diff
@@ -0,0 +1,89 @@
+#include "NamedColor.h"
+
+#include <sstream>
+
+#include <util/StringUtils.h>
+
+NamedColor::NamedColor():
+        paletteIndex{0}, isPaletteColor{false}, name{"Custom Color"}, color{Color(0u)}, colorU16{ColorU16{}} {}
+
+NamedColor::NamedColor(const Color& color): paletteIndex{0}, isPaletteColor{false}, name{"Custom Color"} {
+    this->color = color;
+    this->colorU16 = Util::GdkRGBA_to_ColorU16(Util::argb_to_GdkRGBA(color));
+};
+
+NamedColor& NamedColor::operator=(const NamedColor& other) {
+    if (this == &other)
+        return *this;
+    paletteIndex = other.paletteIndex;
+    isPaletteColor = other.isPaletteColor;
+    name = other.name;
+    color = other.color;
+    colorU16 = other.colorU16;
+    return *this;
+}
+
+NamedColor::NamedColor(NamedColor&& other) noexcept:
+        paletteIndex{other.paletteIndex},
+        isPaletteColor{other.isPaletteColor},
+        name{other.name},
+        color{other.color},
+        colorU16{other.colorU16} {}
+
+NamedColor& NamedColor::operator=(NamedColor&& other) noexcept {
+    if (this == &other)
+        return *this;
+    paletteIndex = other.paletteIndex;
+    isPaletteColor = other.isPaletteColor;
+    name = other.name;
+    color = other.color;
+    colorU16 = other.colorU16;
+    return *this;
+};
+
+
+NamedColor::NamedColor(const NamedColor& other):
+        paletteIndex{other.paletteIndex}, isPaletteColor{other.isPaletteColor}, name{other.name} {
+    this->color = other.color;
+    this->colorU16 = other.colorU16;
+}
+
+NamedColor::~NamedColor(){};
```

Comment:
It would be better to remove all custom defined destructors, copy/move constructors/asignment operators.
Those are created automatically, if all used types also implement those.
Mark the class optionally as final.

Reply:
Makes sense. Will be done :heavy_check_mark: .
Didn't know the `final` specifier before thanks for letting me learn something new :pray: